storage/gcloud/aio/storage/storage.py

```diff
@@ -302,6 +302,38 @@ async def upload_from_filename(self, bucket: str, object_name: str,
             return await self.upload(bucket, object_name, file_object,
                                      **kwargs)
 
+    async def set_temporary_hold(
```

Comment:
It looks like there are a few dozen different aspects of an object which may be patched. What do you think of the relative merits of gradually exposing more and more of these `set_*()` methods vs, say, exposing `.patch(metadata: Dict[str, Any])` directly?

Reply:
I like it — in that case, I can use say `def _set_metadata()` to indicate a private function for now, until it's clear that that the set of fields to patch has grown enough; alternatively, as this grows, we can also use an enum, named arguments etc., as a way to control the vocabulary of fields to those that are actually supported by each Google resource (starting here at Storage).

Reply:
Hmm I guess I'm not sure we need to allowlist the available fields at all? Why not allow users to pass in `metadata={'temporaryHold': True}` directly? Then we won't need to manually build every new feature we want to be able to patch.

I think you can just do `s/_patch_metadata/patch_metadata/` with your current code (and maybe keep some of these defaults such as `headers = None`) and be in a good place -- does being overly proscriptive here win us something I'm missing?

Reply:
No, that's fine — I misunderstood the direction you were going in the original comment.